x/posts/internal/types/media_post.go

```diff
@@ -0,0 +1,260 @@
+	return mp.ParentID
+}
+
+func (mp MediaPost) SetMessage(message string) Post {
+	mp.Message = message
+	return mp
+}
+
+func (mp MediaPost) GetMessage() string {
+	return mp.Message
+}
+
+func (mp MediaPost) CreationTime() time.Time {
+	return mp.Created
+}
+
+func (mp MediaPost) SetEditTime(time time.Time) Post {
+	mp.LastEdited = time
+	return mp
+}
+
+func (mp MediaPost) GetEditTime() time.Time {
+	return mp.LastEdited
+}
+
+func (mp MediaPost) CanComment() bool {
+	return mp.AllowsComments
+}
+
+func (mp MediaPost) GetSubspace() string {
+	return mp.Subspace
+}
+
+func (mp MediaPost) GetOptionalData() map[string]string {
+	return mp.OptionalData
+}
+
+func (mp MediaPost) Owner() sdk.AccAddress {
+	return mp.Creator
+}
+
+// Validate implements Post Validate
+func (mp MediaPost) Validate() error {
+	if err := mp.TextPost.Validate(); err != nil {
+		return err
+	}
+
+	for _, media := range mp.Medias {
+		if err := media.Validate(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (mp MediaPost) Equals(other Post) bool {
+	// Cast and delegate
+	if otherMp, ok := other.(MediaPost); ok {
+		return checkMediaPostEquals(mp, otherMp)
+	}
+
+	return false
+}
+
+// Equals implements Post Equals
+func checkMediaPostEquals(first MediaPost, second MediaPost) bool {
+	if !first.TextPost.Equals(second.TextPost) {
+		return false
+	}
+
+	if len(first.Medias) != len(second.Medias) {
+		return false
+	}
+
+	for index, media := range first.Medias {
+		if media != second.Medias[index] {
+			return false
+		}
+	}
+	return true
+}
+
+// MarshalJSON implements Marshaler
+func (mp MediaPost) MarshalJSON() ([]byte, error) {
+	type temp MediaPost
+	return json.Marshal(temp(mp))
+}
+
+// UnmarshalJSON implements Unmarshaler
+func (mp *MediaPost) UnmarshalJSON(data []byte) error {
+	type mediaPostJSON MediaPost
+	var temp mediaPostJSON
+	if err := json.Unmarshal(data, &temp); err != nil {
+		return err
+	}
+	*mp = MediaPost(temp)
+	return nil
+}
+
+// -------------
+// --- MediaPosts
+// -------------
+
+// TextPosts represents a slice of TextPost objects
+type MediaPosts []MediaPost
+
+// checkPostsEqual returns true iff the p slice contains the same
+// data in the same order of the other slice
+func (mps MediaPosts) Equals(other MediaPosts) bool {
+	if len(mps) != len(other) {
+		return false
+	}
+
+	for index, post := range mps {
+		if !post.Equals(other[index]) {
+			return false
+		}
+	}
+
+	return true
+}
+
+// String implements stringer interface
+func (mps MediaPosts) String() string {
+	bytes, err := json.Marshal(&mps)
+	if err != nil {
+		panic(err)
+	}
+
+	return string(bytes)
+}
+
+// ---------------
+// --- PostMedias
+// ---------------
+
+type PostMedias []PostMedia
+
+func (pms PostMedias) String() string {
+	bytes, err := json.Marshal(&pms)
+	if err != nil {
+		panic(err)
+	}
+
+	return string(bytes)
+}
+func (pms PostMedias) Equals(other PostMedias) bool {
+	for index, postMedia := range pms {
```

Comment:
This is now checking for lengths differences. 

For example, if we have two arrays like 

```
A1 = [M1, M2]
A2 = [M2]
```

This would return true while it should be false. 

Please fix this and add a test case to show that this fails.